api/src/main/java/org/eclipse/microprofile/config/spi/ConfigValue.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.eclipse.microprofile.config.spi;
+
+import java.util.Objects;
+
+/**
+ * The ConfigValue is a metadata object that holds additional information after the lookup of a configuration.
+ * <p>
+ *
+ * Right now, it is able to hold information like the configuration name, value, the Config Source from where
+ * the configuration was loaded, the ordinal of the Config Source and a line number from where the configuration was
+ * read if exists.
+ * <p>
+ *
+ * This is used together with {@link ConfigSource} to expose the Configuration lookup metadata.
+ *
+ * @author <a href="mailto:radcortez@yahoo.com">Roberto Cortez</a>
+ */
+public class ConfigValue {
```

Comment:
You have to be careful when choosing names which already have a pretty established meaning in the mp-config and ConfigJSR area. Like ConfigValue which later got renamed to ConfigAccessor. People will be terribly confused I fear.

Reply:
I'm terrible for naming :D

I didn't come up with the name myself. It was reference like that through our meetings and there are multiple references across other issues.

Do we want to name this ConfigAccessor then?

Reply:
Why not look at work already done. This sadly has a long history with a ton of existing IP.
The original work of accessing config values was actually done in DeltaSpike many years ago (2011). It was also part of the original mp-config work I brought over in 1.0. But back then it was deemed to be 'probably too much in one step'. So we postponed it and reviewed it for ConfigJSR as ConfigValue:
https://github.com/eclipse/ConfigJSR/commit/a5757192d1d11da7e8602e56a337931d6a169541#diff-84c09a90e39a639354bfc30e2100d487

it later got renamed to ConfigAccessor it would be a better fit to what it actually does
https://github.com/eclipse/ConfigJSR/commit/2021eab512c1f5999272d0ac866619868e9d89da#diff-2334b9d3b42caf9078203da0f6f7d82f

Reply:
This was already reviewed and simplified in the current PR. Please check if you agree with the current proposal, regarding the API (only 4 methods for now, to retrieve the property name, value, source from where it was loaded and ordinal). 

Regarding the name, it was always discussed like this (at least in the conversions I've participated). I have no issues in naming it `ConfigAccessor` if everyone else is ok with it.